src/Features/Core/Portable/AddConstructorParametersFromMembers/State.cs

```diff
@@ -38,30 +31,50 @@ private class State
 
             private bool TryInitialize(
                 AddConstructorParametersFromMembersCodeRefactoringProvider service,
-                Document document,
-                TextSpan textSpan,
-                ImmutableArray<ISymbol> selectedMembers,
-                CancellationToken cancellationToken)
+                ImmutableArray<ISymbol> selectedMembers)
             {
                 if (!selectedMembers.All(IsWritableInstanceFieldOrProperty))
                 {
                     return false;
                 }
 
-                this.TextSpan = textSpan;
-                this.SelectedMembers = selectedMembers;
-                this.ContainingType = this.SelectedMembers[0].ContainingType;
+                this.ContainingType = selectedMembers[0].ContainingType;
                 if (this.ContainingType == null || this.ContainingType.TypeKind == TypeKind.Interface)
                 {
                     return false;
                 }
 
-                this.Parameters = service.DetermineParameters(selectedMembers);
+                var parameters = service.DetermineParameters(selectedMembers);
+                this.DelegatedConstructor = service.GetDelegatedConstructorBasedOnParameterNames(this.ContainingType, parameters);
+
+                if (this.DelegatedConstructor != null)
```

Comment:
tiny nit, but i think it would be cleaner to write this as:

```c#
if (this.DeelgatedConstructor == null) {
    return false;
}

// rest of zipping logic
return true;
```